### PR TITLE
Improve PlayerTarget code

### PR DESCRIPTION
**Pull request by JRoy**
Don't know a good title but man that existing code is dumb.

Also resolves https://github.com/EssentialsX/Essentials/discussions/5114

### CODE REVIEW

**Comment by mdcfe** [comment]
What does this even achieve?

**Comment by pop4959** [comment]
Yeah this seems like a regression, this change implies a higher likelihood of stale player references floating around, and also exposes an implementation detail.

The only thing I would potentially agree with is changing this to use a UUID instead of a name, but even that is extremely minor since the target is unlikely to remain relevant beyond a single user session.

**Comment by JRoy** [comment]
made it uuids

File: Essentials/src/main/java/com/earth2me/essentials/PlayerTarget.java [hunk 1]
```diff
@@ -4,15 +4,21 @@
...
-    private final String name;
+    private final UUID uuid;
 
     public PlayerTarget(final Player entity) {
-        this.name = entity.getName();
+        this.uuid = entity.getUniqueId();
+    }
+
+    public UUID getUuid() {
+        return uuid;
```
Comment (pop4959) [inline Essentials/src/main/java/com/earth2me/essentials/PlayerTarget.java]: Implementation detail
```suggestion
```

**Review by pop4959 — CHANGES_REQUESTED** [review]

File: Essentials/src/main/java/com/earth2me/essentials/PlayerTarget.java [hunk 2]
```diff
@@ -4,15 +4,21 @@
 import org.bukkit.Location;
 import org.bukkit.entity.Player;
 
+import java.util.UUID;
+
 public class PlayerTarget implements ITarget {
-    private final String name;
+    private final UUID uuid;
```
Comment (JRoy) [inline Essentials/src/main/java/com/earth2me/essentials/PlayerTarget.java:10]:
```suggestion
    public final UUID uuid;
```

File: Essentials/src/main/java/com/earth2me/essentials/PlayerTarget.java [hunk 3]
```diff
@@ -4,15 +4,18 @@
...
+import java.util.UUID;
+
 public class PlayerTarget implements ITarget {
-    private final String name;
+    public final UUID uuid;
 
     public PlayerTarget(final Player entity) {
-        this.name = entity.getName();
+        this.uuid = entity.getUniqueId();
+    }
```
Comment (JRoy) [inline Essentials/src/main/java/com/earth2me/essentials/PlayerTarget.java:14]:
```suggestion
```

File: Essentials/src/main/java/com/earth2me/essentials/PlayerTarget.java [hunk 4]
```diff
@@ -4,15 +4,17 @@
 import org.bukkit.Location;
 import org.bukkit.entity.Player;
 
+import java.util.UUID;
+
 public class PlayerTarget implements ITarget {
-    private final String name;
+    public final UUID uuid;
```
Comment (pop4959) [inline Essentials/src/main/java/com/earth2me/essentials/PlayerTarget.java]: This is an implementation detail
```suggestion
    private final UUID uuid;
```

**Review by pop4959 — CHANGES_REQUESTED** [review]

**Review by pop4959 — APPROVED** [review]
lgtm

**Pull request merged**